README.md

```diff
@@ -231,19 +229,25 @@ def hidden_states7(x: str) -> list[int]:
     >>> hidden_states7('NNCCCCCCNNRRRRRRN')
     [3, 3, 0, 1, 2, 0, 1, 2, 3, 3, 4, 5, 6, 4, 5, 6, 3]
     """
-    ann = [-1] * len(x)
-    for i, a in enumerate(x):
-        match a:
-            case 'N': ann[i] = 3
-            case 'C' if x[i - 1] != 'C':
-                ann[i] = 0
-            case 'C' if x[i - 1] == 'C':
-                ann[i] = (ann[i - 1] + 1) % 3
-            case 'R' if x[i - 1] != 'R':
-                ann[i] = 4
-            case 'R' if x[i - 1] == 'R':
-                ann[i] = (ann[i - 1] -4 + 1) % 3 + 4
-    return ann
+    hid=[None for i in range(len(x))]
+    for i,v in enumerate(x):
+        if v == 'N':
+            hid[i]=3
+        elif v == 'C' and x[i-1] != 'C':
+            hid[i]=0
+        elif v == 'C' and x[i-1] == 'C':
+            hid[i] = (hid[i-1]+1)%3
+        elif v == 'R' and x[i-1] != 'R':
+            hid[i]=4
+        elif v == 'R' and x[i-1] == 'R':
+            if hid[i-1] == 6:
+                hid[i]=4
+            else:
+                hid[i] = (hid[i-1]+1)%7
```

Comment:
Presumably `hid[i-1]` cannot be larger than 6, as there are only seven states and we number them from zero. You treat `hid[i-1]==6` as a special case, so fi we are here, `hid[i-1]` is at most 5, which mean the modulus will never change the number.